ORE_EEG_2.py

```diff
@@ -5,24 +5,180 @@
+            "params": {
+                "command": "connect",
+                "headset": headset,
+                "mappings": { # under the assumption that the headset is an EPOC Flex
+                    "CMS": "F3",
+                    "DRL": "F5",
+                    "LA": "AF3",
+                    "LB": "AF7",
+                    "RA": "P8"
+                }
+            }
+        }, websocket)
+        response = await send_message({ # authorize the connection
+            "id": 1,
+            "jsonrpc": "2.0",
+            "method": "authorize",
+            "params": {
+                "clientId": os.environ.get('CLIENT_ID'),
+                "clientSecret": os.environ.get('CLIENT_SECRET'),
+                "debit": 1000
+            }
+        }, websocket)
+        if "error" in response:
+            print(response["error"]) # if it gets here, probably didn't set up env variables correctly
+            return
+        cortex_token = response["result"]["cortexToken"]
+        response = await send_message({
+            "id": 1,
+            "jsonrpc": "2.0",
+            "method": "createSession",
+            "params": {
+                "cortexToken": cortex_token,
+                "headset": headset,
+                "status": "open"
+            }
+        }, websocket)
+        session_id = response["result"]["id"]
+
+        await send_message({
+            "id": 1,
+            "jsonrpc": "2.0",
+            "method": "updateSession",
+            "params": {
+                "cortexToken": cortex_token,
+                "session": session_id,
+                "status": "active"
+            }
+        }, websocket)
+
+        response = await send_message({
+            "id": 1,
+            "jsonrpc": "2.0",
+            "method": "createRecord",
+            "params": {
+                "cortexToken": cortex_token,
+                "session": session_id,
+                "title": "Alljoined Cortex Recording Session"
+            }
+        }, websocket)
+        record_id = response["result"]["record"]["uuid"]
+
+        headset_info["headset"] = headset
+        headset_info["cortex_token"] = cortex_token
+        headset_info["session_id"] = session_id
+        headset_info["record_id"] = record_id
+        headset_info["record_ids"] = [record_id]
+
+async def teardown_eeg():
+    async with websockets.connect("wss://localhost:6868", ssl=ssl_context) as websocket:
+        response = await send_message({
+            "id": 1,
+            "jsonrpc": "2.0",
+            "method": "exportRecord",
+            "params": {
+                "cortexToken": headset_info["cortex_token"],
+                "folder": "/tmp/edf",
+                "format": "EDF",
+                "recordIds": headset_info["record_ids"],
+                "streamTypes": [
+                    "EEG",
+                    "MOTION"
+                ]
+            }
+        }, websocket)
+        print(response)
+        # wait for warning 18
```

Comment:
what is warning 18?

Reply:
it's for [successfully saved record](https://emotiv.gitbook.io/cortex-api/warning-objects)
> You must [stop the record](https://emotiv.gitbook.io/cortex-api/records/stoprecord) before you can export it. If you want to export a record immediately after you stop it then you must  wait for the [warning 18](https://emotiv.gitbook.io/cortex-api/warning-objects#warning-codes) before you try to export.

[from documentation](https://emotiv.gitbook.io/cortex-api/records/exportrecord)

Reply:
gotcha. Maybe link to the documentation and copy paste this:

`The record data has been successfully saved. Cortex sends this warning when a record is done with some long-run post processing after record is stopped.`